# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prim_store/mod.rs
# Commit: 32951c9cc186
# Full Hash: 32951c9cc186f044c7a59e0b132339139dfb9713
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2019-09-18 09:56:40
# Regressor Bug: 1570081
# File Overlap Count: 1
# Description:
#   Bug 1570081 - Support changing the blob image visible rect in WebRender. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D39911
# ==============================================================================

diff -r 37598891cea6 -r 32951c9cc186 gfx/wr/webrender/src/prim_store/mod.rs
--- a/gfx/wr/webrender/src/prim_store/mod.rs	Tue Sep 17 09:20:04 2019 +0000
+++ b/gfx/wr/webrender/src/prim_store/mod.rs	Tue Sep 17 09:12:26 2019 +0000
@@ -2289,9 +2289,12 @@
                             frame_state.gpu_cache,
                         );
                     }
-                    Some(ImageProperties { descriptor, tiling: Some(tile_size), .. }) => {
+                    Some(ImageProperties { tiling: Some(tile_size), visible_rect, .. }) => {
                         image_instance.visible_tiles.clear();
-                        let device_image_rect = DeviceIntRect::from_size(descriptor.size);
+                        // TODO: rename the blob's visible_rect into something that doesn't conflict
+                        // with the terminology we use during culling since it's not really the same
+                        // thing.
+                        let active_rect = visible_rect;
 
                         // Tighten the clip rect because decomposing the repeated image can
                         // produce primitives that are partially covering the original image
@@ -2344,7 +2347,7 @@
                             let tiles = crate::image::tiles(
                                 &layout_image_rect,
                                 &visible_rect,
-                                &device_image_rect,
+                                &active_rect,
                                 tile_size as i32,
                             );
 